### PR TITLE
Allowing user to specify full URL to social page in widget

### PR DESCRIPTION
**Pull request by howtoaddict**
Previously user could only specify username on Social media. This caused
problems with some sites like Youtube... on which you can have both User
(http://youtube.com/u/Username) and Channel
(http://youtube.com/c/Channel).

### CODE REVIEW

**Comment by jeherve** [comment]
Fixes #6010

File: modules/widgets/social-media-icons.php [hunk 1]
```diff
@@ -137,8 +137,22 @@ public function widget( $args, $instance ) {
...
-				'googleplus' === $service
+
+			/** Check if full URL entered in configuration, use it instead of tinkering **/
+			if ( substr( $username, 0, 5 ) === "http:"
+				|| substr( $username, 0, 6 ) === "https:"
+			   ) {			
+				$html[ $index ] =
+					'<a href="' . $username
+					. '" class="genericon genericon-' . $service . '" target="_blank"><span class="screen-reader-text">'
+					. 'View our profile on '. $service_name
```
Comment (eliorivero) [inline modules/widgets/social-media-icons.php]: This text must be made available for translation like this:
```
/* Translators: the placeholder is a social network name. */
. esc_html( sprintf( __( 'View our profile on %s', 'jetpack' ), $service_name ) )
```

File: modules/widgets/social-media-icons.php [hunk 2]
```diff
@@ -137,8 +137,22 @@ public function widget( $args, $instance ) {
 				continue;
 			}
 			$index += 10;
-			if (
-				'googleplus' === $service
+
+			/** Check if full URL entered in configuration, use it instead of tinkering **/
+			if ( substr( $username, 0, 5 ) === "http:"
+				|| substr( $username, 0, 6 ) === "https:"
```
Comment (eliorivero) [inline modules/widgets/social-media-icons.php]: This should be replaced with
```
if ( preg_match( '#^https?://#', $username ) ) {
```

**Comment by howtoaddict** [comment]
@eliorivero updated

**Comment by georgestephanis** [comment]
Would this change let someone paste in a link to "http://hackmysite.tld" and have it show with the social media icon of the relevant service?  I'm not sure if it may be wise to build in some sort of domain name verification, or if that's being overly and unnecessarily cautious.

**Comment by eliorivero** [comment]
Yeah, George is right. We shouldn't allow people to do evil. Maybe something like this would be enough?
```php
is_valid_social_network( $url ) {
    return in_array( parse_url( $url, PHP_URL_HOST ), array(
        'twitter.com',
        'facebook.com',
        etc etc
    ) );
}
```

**Comment by howtoaddict** [comment]
@georgestephanis @eliorivero yeah... like you can't use other components in Jetpack to do exactly that - put COMPLETELY custom HTML.

I've invested time trying to help and this is obviously beneficial and requested feature. If you guys don't want to add it don't merge it. I'm giving up on this pull request.

**Comment by dereksmart** [comment]
Right, @howtoaddict has a very valid point.  There are many many places throughout WordPress core and Jetpack features that allow inserting custom HTML.  Someone could do the same thing with a simple WordPress text widget.  I don't think it's very realistic to expect us to filter all of these instances.  

@eliorivero's suggestion does seem pretty reasonable in theory, but that would require us having to create an index of full/shorthand/custom social media URLs that are used everywhere.  IMO more effort than it's worth.

**Comment by eliorivero** [comment]
Let's 🐑 this. Thanks @howtoaddict 👍

**Comment by lepipele** [comment]
Thanks @dereksmart @georgestephanis @eliorivero and sorry if I was too direct ... I'm swamped with my regular job and code arguing gets me into edgy mood.

**Comment by zinigor** [comment]
Closing this in favour of #6508, thanks for the idea and the implementation!

**Pull request closed without merging**